src/main/webapp/app/entities/flair-bi/filter/filter-element-grpc.component.js

```diff
@@ -345,8 +345,17 @@
         }
 
         function addFilterInIframeURL() {
-            var filters = filterParametersService.get();
-            filterParametersService.setFilterInIframeURL(filters,vm.iframes,vm.dimension);
+            Views.getCurrentEditState({
+                id: $stateParams.id
+            },
+                function (result, headers) {
+                    vm.iFrames = result.visualMetadataSet.filter(function (item) {
+                        return item.metadataVisual.name === IFRAME.iframe;
+                    })
+                    var filters = filterParametersService.get();
```

Comment:
This line is not required... You can get the filters object inside filter parameter service. So no need to pass filters in this function setFilterInIframeURL